python/src/aac/plugins/first_party/lsp_server/providers/goto_definition_provider.py

```diff
@@ -77,9 +77,7 @@ def get_definition_location(self, name: str) -> list[Location]:
                 locations.extend(self.get_enum_value_lexeme_location(definition_to_find, name))
 
         if SymbolType.ROOT_KEY in symbol_types:
-            definition_to_find = language_context.get_root_keys_definition()
-            if definition_to_find:
-                locations.extend(self.get_root_key_definition_lexeme_location(definition_to_find, name))
+            logging.warn("Legacy Root key no longer exists and has no lexeme.")
```

Comment:
Like this as a way of assisting the end user to know that a breaking change has been made.

Would this be a check that could go away in later versions because `ROOT_KEY` is no longer a symbol type for the LSP to worry about? Or would it be reworked to reference the new `root` entry and take you to where that definition resides?

I've not messed with the VSCode LSP functionality, so not 100% on how it is utilized or how it might need expanded to incorporate the new take on `root`, if it does need to be.

Reply:
I'm also a novice with the VSCode LSP functionality and wanted to make a few updates here as possible while ensuring all tests pass.  I had experimented with removing `SymbolType.ROOT_KEY` all together but that cascaded rapidly in ways I didn't fully understand.  In the end, I felt that a good incremental step would be to pop out a warning if the logic was executed.

Reply:
Can easily believe that rippled because there was another spot within the LSP plugin that had a check using `SymbolType.ROOT_KEY`. Agree this mid-step is very useful in keeping a user aware and giving us the time to identify and handle the ripples in a more dedicated manner.